src/background/utils/db.js

```diff
@@ -158,6 +161,13 @@ preInitialize.push(async () => {
       return map;
     }, {}),
   });
+  // Switch defaultInjectInto from `page` to `auto` when upgrading VM2.12.7 or older
+  if (version !== lastVersion
+  && ua.isFirefox
+  && data.options?.defaultInjectInto === INJECT_PAGE
+  && `${lastVersion}`.replace(/\b\d\b/g, s => `0${s}`) <= '02.12.07') {
```

Comment:
Can we use `compareVersion` here?

Reply:
Done, thanks. I've also just added an indication of the failure and a clickable action to fix it. The indication is a red strike-through, intentionally weird to be noticeable (too weird?). You can try it on this github page when VM's default injection mode is `page`.